src/name/resolve.js

```diff
@@ -13,6 +15,19 @@ module.exports = (send) => {
       opts = {}
     }
 
+    // normalize PeerIDs to Base58btc, so it works with go-ipfs <=0.4.22
+    if (typeof (args) === 'string') {
+      try {
+        const path = args.split('/')
+        if (path.length > 2) {
+          path[2] = multihash.toB58String(new CID(path[2]).multihash)
+          args = path.join('/')
+        }
+      } catch (err) {
+        // noop
+      }
+    }
```

Comment:
is there a better way? :/

Reply:
All we need to do is stringify `args` if it is a CID. We will skip the `interface-ipfs-core` test here until go-ipfs supports it.

Reply:
We don't stringify CIDs anywhere else, removed this hack and added TODO to unskip mentioned test when go-ipfs implements this.

Reply:
Ah interesting, I guess the querystring module calls `toString` on the CID then...